facade/src/main/java/org/jboss/pnc/facade/providers/SCMRepositoryProviderImpl.java

```diff
@@ -71,9 +117,120 @@ public SCMRepositoryProviderImpl(RepositoryConfigurationRepository repository,
      * @param <T>
      */
     private <T> void addToListIfStringNotNullAndNotEmpty(List<T> list, String str, T item) {
-
         if (str != null && !str.isEmpty()) {
             list.add(item);
         }
     }
+
+    @Override
+    public RepositoryCreationResponse createSCMRepository(String scmUrl, Boolean preBuildSyncEnabled, Consumer<Integer>... consumers) {
+        log.trace("Received request to start RC creation with url autodetect: " + scmUrl + " (sync enabled? " + preBuildSyncEnabled + ")");
+        if(StringUtils.isEmpty(scmUrl)) throw new InvalidEntityException("You must specifyt the SCM URL.");
+
+        if(scmUrl.contains(config.getInternalScmAuthority())){
+            SCMRepository scmRepository = getInternalRepository(scmUrl);
```

Comment:
Technically speaking the internal repository is always sychnronized to itselve. I think just ignoring the field is fine.

Reply:
But it could lead to misleading data and cause confusion. If I was a user and seen some RC(only internal specified) with preBuildSync enabled and some disabled. I could start making assumptions that this syncing is making changes to my repository and possibly compromise my product. I'd have the check there so no one would ask unnecessary questions while panicking a little and leave them with a feeling of certainty. It makes our data cleaner too.

Reply:
All the RC with only internal URL specified has the SCM disabled. The value is ignored in the request and set to false.

Reply:
Then I have no objections!